lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -108,16 +108,17 @@ class ConstOpConversion : public OpRewritePattern<tosa::ConstOp> {
   }
 };
 
-class Conv2DOpConversion : public OpConversionPattern<tosa::Conv2DOp> {
-  using OpConversionPattern<tosa::Conv2DOp>::OpConversionPattern;
+template <typename SrcOp>
+class Conv2DOpConversion : public OpConversionPattern<SrcOp> {
+  using OpConversionPattern<SrcOp>::OpConversionPattern;
 
 public:
-  Conv2DOpConversion(MLIRContext *ctx)
-      : OpConversionPattern<tosa::Conv2DOp>(ctx) {}
+  Conv2DOpConversion(MLIRContext *ctx, StringRef funcName)
```

Comment:
```suggestion
  ConvOpConversion(MLIRContext *ctx, StringRef funcName)
```

If it's not only for a 2D conv anymore, this should be renamed.

Reply:
`depthwise_conv2d` is still a `conv2d` :P

Reply:
Yes, but the name maps to closely to `Conv2DOp` which is somehow misleading.

Reply:
I renamed it to `GenericConvOpConversion`